Dockerfile

```diff
@@ -1,6 +1,21 @@
 #####################################################################################################
+# Build arguments
+ARG BUILDER_IMAGE=quay.io/projectquay/golang:1.23
+ARG RUNTIME_IMAGE=registry.access.redhat.com/ubi9-minimal:9.4
+# note: update origin-cli-artifacts from `latest` to an appropriate OCP version during release e.g `4.18`
+ARG ORIGIN_CLI_IMAGE=quay.io/openshift/origin-cli-artifacts:latest
+
+# Assume x86 unless otherwise specified
+ARG GOARCH="amd64"
+
 # Build the binaries
-FROM registry.access.redhat.com/ubi9/go-toolset:1.22.5-1730550521 as builder
```

Comment:
@mresvanis I can't recall why we started using the go-toolset image. It seems slow on releasing newer golang versions. Can you double check if Steven's change is good?

Reply:
I found the change here in an older PR: https://github.com/openshift-kni/lifecycle-agent/pull/65

and yeah it seems like go1.23 is not available yet in ubi.

Also note that for Konflux we will be using the openshift golang builder but that will be in a separate PR: https://github.com/openshift-kni/lifecycle-agent/pull/763/files#diff-e5b5fa6deab8f9cda1e5725b06bd8d766482779861b02c95a47669a7ff8a844e